src/Authenticator/AuthenticatorCollection.php

```diff
@@ -49,26 +49,21 @@ public function __construct(IdentifierCollection $identifiers, array $config = [
     /**
      * Creates authenticator instance.
      *
-     * @param object|string $class Authenticator class.
+     * @param \Authentication\Authenticator\AuthenticatorInterface|string $class Authenticator class.
      * @param string $alias Authenticator alias.
      * @param array $config Config array.
      * @return \Authentication\Authenticator\AuthenticatorInterface
      * @throws \RuntimeException
      */
     protected function _create(object|string $class, string $alias, array $config): AuthenticatorInterface
     {
-        $authenticator = $class;
         if (is_string($class)) {
-            $authenticator = new $class($this->_identifiers, $config);
-        }
-        if (!($authenticator instanceof AuthenticatorInterface)) {
-            throw new RuntimeException(sprintf(
-                'Authenticator must implement `%s`.',
-                AuthenticatorInterface::class
-            ));
+            /** @var class-string<\Authentication\Authenticator\AuthenticatorInterface> $class */
+            return new $class($this->_identifiers, $config);
         }
 
-        return $authenticator;
+        /** @var \Authentication\Authenticator\AuthenticatorInterface $class */
+        return $class;
```

Comment:
psalm uses the `object|string` inferred type instead of the annotated type so have to specify $class as \Authentication\Authenticator\AuthenticatorInterface